grails-app/services/org/pih/warehouse/order/OrderSummaryService.groovy

```diff
@@ -0,0 +1,441 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.order
+
+import org.pih.warehouse.data.DataService
+import org.pih.warehouse.jobs.RefreshOrderSummaryJob
+
+
+/**
+ * TODO: This OrderSummaryService is temporary (or to be expanded) solution for performance issues with
+ *  order summary helper views during fetching and reloading data. In next iteration we should migrate
+ *  from using views to use queries and/or create stored procedures (imho we should get rid of views and leave
+ *  only the materialized views)
+ * */
+class OrderSummaryService {
+
+    DataService dataService
+
+    String getOrderItemStatusSelect(String orderId) {
+        return """
+            SELECT
+                order_id,
+                order_number,
+                IFNULL(MAX(quantity_ordered), 0)    AS quantity_ordered,
+                IFNULL(SUM(quantity_shipped), 0)    AS quantity_shipped,
+                order_item_id
+            FROM (
+                SELECT
+                    `order`.id                  AS order_id,
+                    `order`.order_number        AS order_number,
+                    order_item.id               AS order_item_id,
+                    CASE
+                        WHEN order_item.order_item_status_code = 'CANCELED' THEN 0
+                        ELSE IFNULL(MAX(order_item.quantity), 0)
+                    END AS quantity_ordered,
+                    CASE
+                        WHEN shipment.current_status IN ('SHIPPED', 'PARTIALLY_RECEIVED', 'RECEIVED') THEN SUM(shipment_item.quantity / order_item.quantity_per_uom)
+                        ELSE 0
+                    END AS quantity_shipped
+                FROM `order`
+                    LEFT OUTER JOIN order_item ON order.id = order_item.order_id
+                    LEFT OUTER JOIN order_shipment ON order_item.id = order_shipment.order_item_id
+                    LEFT OUTER JOIN shipment_item ON shipment_item.id = order_shipment.shipment_item_id
+                    LEFT OUTER JOIN shipment ON shipment.id = shipment_item.shipment_id
+                WHERE `order`.order_type_id = 'PURCHASE_ORDER'
+                    AND `order`.id = '${orderId}'
```

Comment:
so are these queries faster because you're pre-filtering only for the order id that you need?

Reply:
Yes, but on top of that, there is no need to build entire view